clients/adapter-amqp/src/main/java/org/eclipse/hono/adapter/client/registry/amqp/ProtonBasedDeviceRegistrationClient.java

```diff
@@ -76,8 +76,12 @@ public ProtonBasedDeviceRegistrationClient(
             final ProtocolAdapterProperties adapterConfig,
             final Cache<Object, RegistrationResult> responseCache) {
 
-        super(connection, samplerFactory, adapterConfig, new CachingClientFactory<>(
-                connection.getVertx(), RequestResponseClient::isOpen), responseCache);
+        super(connection,
+                samplerFactory,
+                adapterConfig.isDefaultsEnabled(),
+                adapterConfig.isJmsVendorPropsEnabled(),
```

Comment:
> In this case here, this would mean whether the device registry implementation is JMS based and would actually need to read such properties. FMPOV, the latter isn't the case with our device registry APIs (the contentType being always JSON or empty anyway, depending on the operation).

I guess you are right here. We took great care to allow registry services being implemented using JMS without requiring the vendor properties being used.

> With the other comment above in mind, I think it would even be better to remove both new constructor params from AbstractServiceClient altogether and only use them directly in ProtonBasedDownstreamSender and ProtonBasedCommandResponseSender (usage is missing there still).

I guess you are right here as well. Great feedback :+1: